src/sentry/eventstore/snuba/backend.py

```diff
@@ -108,9 +110,62 @@ def __get_events(
         should_bind_nodes=False,
     ):
         assert filter, "You must provide a filter"
-        cols = self.__get_columns(additional_columns)
+        cols = self.__get_columns()
         orderby = orderby or DESC_ORDERING
 
+        # This is an optimization for the Group.filter_by_event_id query where we
+        # have a single event ID and want to check all accessible projects for a
+        # direct hit. In this case it's usually faster to go to nodestore first.
+        if (
+            filter.event_ids
+            and filter.project_ids
+            and len(filter.event_ids) * len(filter.project_ids) < min(limit, NODESTORE_LIMIT)
```

Comment:
I remember once we ran into a case where one field in filter was supposed to contain a list but could also contain a string. Do you remember that case? It was in the early eventstore PRs. If it was one of these and it is still a thing, this can break.

Reply:
Hm, I don't recall exactly, that might have been the case for some filter_keys at one point but not in the eventstore Filter. Looking at https://github.com/getsentry/sentry/blob/master/src/sentry/utils/snuba.py#L304-L331 it seems like that query would not work if it was the case.